tests/fixture/testnet/local/node.go

```diff
@@ -0,0 +1,295 @@
+	// Ensure a stale process context file is removed so that the
+	// creation of a new file can indicate node start.
+	if err := os.Remove(n.GetProcessContextPath()); err != nil && !os.IsNotExist(err) {
+		return fmt.Errorf("failed to remove stale process context file: %w", err)
+	}
+
+	execPath := n.ExecPath
+	if len(execPath) == 0 {
+		execPath = defaultExecPath
+	}
+
+	cmd := exec.Command(execPath, "--config-file", n.GetConfigPath())
+	if err := cmd.Start(); err != nil {
+		return err
+	}
+
+	go func() {
+		if err := cmd.Wait(); err != nil {
+			if err.Error() != "signal: killed" {
+				_, _ = fmt.Fprintf(w, "node %q finished with error: %v\n", n.NodeID, err)
+			}
+		} else {
+			_, _ = fmt.Fprintf(w, "node %q finished: %v\n", n.NodeID, cmd.ProcessState.String())
+		}
+	}()
+
+	if _, err := fmt.Fprintf(w, "Started %s\n", n.NodeID); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (n *LocalNode) Stop() error {
+	pid := n.PID
+	if pid == 0 {
+		// Nothing to do
+		return nil
+	}
+	proc, err := os.FindProcess(pid)
+	if err != nil {
+		return fmt.Errorf("failed to find process to stop: %w", err)
+	}
+	// Sending 0 will not actually send a signal but will perform
+	// error checking. If no error is returned, the process is still
+	// running.
+	if err := proc.Signal(syscall.Signal(0)); err != nil {
+		if strings.Contains(err.Error(), "os: process already finished") {
+			return nil
+		}
+		return fmt.Errorf("process.Signal(0) on pid %d returned: %w", pid, err)
+	}
+	if err := proc.Signal(syscall.SIGTERM); err != nil {
+		return fmt.Errorf("failed to send SIGTERM to pid %d: %w", pid, err)
+	}
+	return nil
+}
+
+func (n *LocalNode) IsHealthy(ctx context.Context) (bool, error) {
+	// Ensure the API URI is available
+	if len(n.URI) == 0 {
+		if err := n.ReadProcessContext(); err != nil {
+			return false, fmt.Errorf("failed to read process context: %w", err)
+		}
+		if len(n.URI) == 0 {
+			return false, nil
+		}
+	}
+
+	// Check that the process is still alive
+	proc, err := os.FindProcess(n.PID)
+	if err != nil {
+		return false, fmt.Errorf("failed to find process: %w", err)
+	}
+	err = proc.Signal(syscall.Signal(0))
+	switch err {
+	case nil:
+		// Process is alive
+	case syscall.ESRCH:
+		// Process is dead
+		return false, errors.New("process not running")
+	default:
+		return false, errors.New("process running but query operation not permitted")
+	}
+
+	// Check that the node is reporting healthy
+	health, err := health.NewClient(n.URI).Health(ctx, nil)
+	if err != nil {
+		switch t := err.(type) {
+		case *net.OpError:
+			if t.Op == "read" {
+				// Connection refused - potentially recoverable
+				return false, nil
+			}
+		case syscall.Errno:
+			if t == syscall.ECONNREFUSED {
+				// Connection refused - potentially recoverable
+				return false, nil
+			}
+		}
+		// Assume all other errors are not recoverable
+		return false, err
+	}
+	return health.Healthy, nil
```

Comment:
nit reduce indentation: 
```suggestion
	if err == nil {
		return health.Healthy, nil
	}

	switch t := err.(type) {
	case *net.OpError:
		if t.Op == "read" {
			// Connection refused - potentially recoverable
			return false, nil
		}
	case syscall.Errno:
		if t == syscall.ECONNREFUSED {
			// Connection refused - potentially recoverable
			return false, nil
		}
	}
	// Assume all other errors are not recoverable
	return false, err
```